decentraland/unity-renderer
# Feat: info button for settings UI

## What does this PR change?
Closes #3629 

Related to #3629 feature:

- Created info-button and tooltip prefabs with accordance to the design (see #3629)
- Extended `SettingsControlView` and `SettingsControlModel` with support for `flagsThatOvverideMe`.
- Implemented required logic to make some settings overridden, which greyed out control, but not the tittle of the settings, near to which info-button provided (click on this button will show Tooltip). Position of the info-button is auto-adjusted to the text width of the tittle.
- Created `TooltipComponentView`/`Model`, to provide possibility for setting up tooltip message. It includes functionality, that hides tooltip when user clicks outside. 

Other changes:
- Created base/template prefab for `SettingsControlTemplate`'s and inherit specific prefabs from it - for `DropDown`, `Slider`, `Toogle `and `SpinBox`.
- fixed parameters being in PascalCase in `.editorconfig` (now is camelCase)

## How to test the changes?

1. Go to: https://play.decentraland.zone/?renderer-branch=feat/info-button-for-settings-ui&realm=mgoldman.dcl.eth
2. Open `Settings UI` (in Explore menu)
3. Observe that `DYNAMIC SKYBOX` setting is grayed out, but its tittle is not.
4. Observe `info-button` at the right of the `DYNAMIC SKYBOX` tittle. 
5. Click on this `info-button` and observe `tooltip `_"This settings is being controlled by the creator"_ popup above it. 
6. Click on the `tooltip `- nothing should happen. Click outside of it - `tooltip `should disappear

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
